pkg/asset/agent/common.go

```diff
@@ -1,18 +1,50 @@
 package agent
 
 import (
+	"github.com/openshift/installer/pkg/types"
 	"github.com/openshift/installer/pkg/types/baremetal"
 	"github.com/openshift/installer/pkg/types/none"
 	"github.com/openshift/installer/pkg/types/vsphere"
+
+	hiveext "github.com/openshift/assisted-service/api/hiveextension/v1beta1"
 )
 
-// SupportedPlatforms lists the supported platforms for agent installer
-var SupportedPlatforms = []string{baremetal.Name, vsphere.Name, none.Name}
+// SupportedInstallerPlatforms lists the supported platforms for agent installer
+func SupportedInstallerPlatforms() []string {
+	return []string{baremetal.Name, vsphere.Name, none.Name}
+}
+
+var supportedHivePlatforms = []hiveext.PlatformType{
+	hiveext.BareMetalPlatformType,
+	hiveext.VSpherePlatformType,
+	hiveext.NonePlatformType,
+}
+
+// SupportedHivePlatforms lists the supported platforms for AgentClusterInstall
+func SupportedHivePlatforms() []string {
```

Comment:
Lint had some complaints
pkg/asset/agent/common.go:12:81: Comment should end in a period (godot)
// SupportedInstallerPlatforms lists the supported platforms for agent installer
                                                                                ^
pkg/asset/agent/common.go:23:80: Comment should end in a period (godot)
// SupportedHivePlatforms lists the supported platforms for AgentClusterInstall
                                                                               ^
pkg/asset/agent/common.go:25:2: Consider preallocating `platforms` (prealloc)
	var platforms []string